packages/fleet/lib/models/nodes.ts

```diff
@@ -215,20 +250,53 @@ export async function fail(
     db: knex.Knex,
     props: {
         nodeId: number;
-        error: string;
+        reason: string;
+    }
+): Promise<Result<Node>> {
+    return transitionTo(db, { nodeId: props.nodeId, newState: 'ERROR', reason: props.reason });
+}
+
+export async function register(
+    db: knex.Knex,
+    props: {
+        nodeId: number;
+        url: string;
+    }
+): Promise<Result<Node>> {
+    return transitionTo(db, { nodeId: props.nodeId, newState: 'RUNNING', url: props.url });
+}
+
+export async function idle(
+    db: knex.Knex,
+    props: {
+        nodeId: number;
+    }
+): Promise<Result<Node>> {
+    return transitionTo(db, { nodeId: props.nodeId, newState: 'IDLE' });
+}
+
+export async function remove(
+    db: knex.Knex,
+    props: {
+        nodeId: number;
     }
 ): Promise<Result<Node>> {
     try {
-        const updated = await db
-            .from<DBNode>(NODES_TABLE)
-            .where({ id: props.nodeId })
-            .update({ state: 'ERROR', error: props.error, last_state_transition_at: new Date() })
-            .returning('*');
-        if (!updated?.[0]) {
-            return Err(new FleetError(`node_fail_error`, { context: props }));
+        // only TERMINATED and ERROR nodes can be deleted
+        const getNode = await get(db, props.nodeId);
+        if (getNode.isErr()) {
+            return getNode;
+        }
+        if (!['TERMINATED', 'ERROR'].includes(getNode.value.state)) {
+            return Err(new FleetError(`node_delete_non_terminated`, { context: { nodeId: props.nodeId } }));
+        }
+
+        const deleted = await db.from<DBNode>(NODES_TABLE).where('id', props.nodeId).del().returning('*');
```

Comment:
there is no need for returning, just checking row_count should be enough no?

Reply:
the idea was to have every function to return a `Result<Node>`. I agree it is not necessary for deletion but it is more for consistency